Remove ExPlat from the project. Replace it with `experimentation` library from Tracks.

wzieba opened the pull request:
Closes: [AINFRA-661: Remove FluxC dependency on Track's `experimentation`](https://linear.app/a8c/issue/AINFRA-661/remove-fluxc-dependency-on-tracks-experimentation)

## Description
<!-- Describe the changes, why they are needed, and how they address the issue -->

**Update**: this PR removes `ExPlat`/`experimentation` from the project, as there are no running experiments.

This PR is a twin of https://github.com/woocommerce/woocommerce-android/pull/14128. 

The idea here is to replace internal experimentation logic with [`experimentation` library](https://github.com/Automattic/Automattic-Tracks-Android/tree/trunk/experimentation).

The historical context is following:
- initially, `ExPlat` was shared via FluxC between WordPress and WooCommerce
- then, we vendored FluxC into projects
- then, we extracted `experimentation` from WooCommerce's FluxC and replaced it there
- and now, does the same, but for WordPress/Jetpack

## Testing instructions
<!-- Consider listing specific testing steps for helping reviewers evaluate the efficacy of the changes -->

<!--

Consider testing the following *yourself* before requesting a review:

- Compatibility with WordPress.com sites and self-hosted Jetpack sites
- Both portrait and landscape orientations
- Light and dark modes
- Dynamic font sizes: larger, smaller and bold text
- High contrast mode
- Screen reader experience (e.g., Talkback)
- Languages with large words or with letters/accents not frequently used in English
- Right-to-left languages layout
- Multi-tasking: split view and slide over

-->

Actually - not needed. The `experimentation` library was tested during development. As there are no running experiments in WordPress/Jetpack, we can't meaningfully test this PR.

<!--

Use these emoji to improve the code review

❤️ - Praise. Compliments about great solution.
⛏ - Nitpick️. Not high importance. Usually, short one-liners.
💡 - Idea. Alternative solution or improvement.
❓ - Question. Requires some explanation or more context.
⚠️ - Warning. Should be addressed in this PR or in some follow-up PR.
🛑 - Blocker. Must be addressed before PR gets merged.

-->

wzieba committed:
Remove ExPlat logic from the project. Replace it with `experimentation` library from Tracks.

wzieba committed:
Provide `AppLogWrapper#e` with `throwable` directly

Otherwise unit tests fail because Mockito has a problem with the synthethic Kotlin method (doesn't see it)

wzieba committed:
Fix static code analysis issues

wzieba commented on WordPress/src/main/java/org/wordpress/android/modules/ExperimentModule.kt:
```diff
@@ -1,17 +1,39 @@
+        appLogWrapper: AppLogWrapper,
+        @Named(APPLICATION_SCOPE) appScope: CoroutineScope,
+    ) = VariationsRepository.create(
+        platform = "wpandroid",
```
From https://github.com/wordpress-mobile/WordPress-Android/blob/69ed046f53ef8b0b01cbb5ea8b86327be21a540e/libs/fluxc/src/main/java/org/wordpress/android/fluxc/store/ExperimentStore.kt#L89

wzieba commented on WordPress/src/main/java/org/wordpress/android/AppInitializer.kt:
```diff
@@ -663,8 +661,8 @@ class AppInitializer @Inject constructor(
-            // Force a refresh if user has logged in. This can be removed once we start using an anonymous ID.
-            exPlat.forceRefresh()
+            // Possible way to initialize variations repository. Not tested, verify when working on experiments.
+            // variationsRepository.initialize("${tracker.anonID}")
```
I don't implement it because the project doesn't run any experiments, and what we consider "experiment id" can vary from use case. This is just a hint where we should likely initialize `VariationsRepository`.

wzieba commented on WordPress/src/main/java/org/wordpress/android/AppInitializer.kt:
```diff
@@ -934,11 +931,6 @@ class AppInitializer @Inject constructor(
 
             // Let's migrate the old editor preference if available in AppPrefs to the remote backend
             SiteUtils.migrateAppWideMobileEditorPreferenceToRemote(accountStore, siteStore, dispatcher)
-            if (!firstActivityResumed) {
-                // Since we're force refreshing on app startup, we don't need to try refreshing again when starting
-                // our first Activity.
-                exPlat.refreshIfNeeded()
-            }
```
Intended removal. The `experimentation` library will handle per-session cache. There's no need to refresh the experiments on app resume.

jkmassel reviewed (approved):
LGTM, but if we're not doing any experiments right now would it make more sense to just remove without adding a replacement until we need it?

wzieba committed:
Remove `experimentation` from the project

There are no currently running experiments, so this integration is redundant. https://github.com/wordpress-mobile/WordPress-Android/pull/22354#pullrequestreview-3479769334
